src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractReferenceFinder.cs

```diff
@@ -564,6 +564,42 @@ static bool IsRelevantDocument(SyntaxTreeIndex syntaxTreeInfo)
             }
         }
 
+        protected Task<ImmutableArray<FinderLocation>> FindReferencesInElementAccessExpressionsAsync(
+            ISymbol symbol,
+            Document document,
+            SemanticModel semanticModel,
+            bool isFindMethod,
+            CancellationToken cancellationToken)
+        {
+            return FindReferencesInDocumentAsync(symbol, document, IsRelevantDocument, CollectMatchingReferences, cancellationToken);
+
+            static bool IsRelevantDocument(SyntaxTreeIndex syntaxTreeInfo)
+                => syntaxTreeInfo.ContainsElementAccessExpression;
+
+            void CollectMatchingReferences(ISymbol originalUnreducedSymbolDefinition, SyntaxNode node,
+                ISyntaxFactsService syntaxFacts, ISemanticFactsService semanticFacts, ArrayBuilder<FinderLocation> locations)
+            {
+                ISymbol elementAccessMethodOrProperty;
+                if (isFindMethod)
+                {
+                    elementAccessMethodOrProperty = semanticFacts.GetElementAccessMethod(semanticModel, node);
+                }
+                else
+                {
+                    elementAccessMethodOrProperty = semanticFacts.GetElementAccessCountableProperty(semanticModel, node);
+                }
+
+                if (Matches(elementAccessMethodOrProperty, originalUnreducedSymbolDefinition))
+                {
+                    var location = node.GetFirstToken().GetLocation();
+                    var symbolUsageInfo = GetSymbolUsageInfo(node, semanticModel, syntaxFacts, semanticFacts, cancellationToken);
+
+                    locations.Add(new FinderLocation(node, new ReferenceLocation(
+                        document, alias: null, location, isImplicit: true, symbolUsageInfo, GetAdditionalFindUsagesProperties(node, semanticModel, syntaxFacts), CandidateReason.None)));
```

Comment:
do we get the right symbolUsageInfo and FindUsagesProperties here? what do we show the user?